osu.Game/Screens/OnlinePlay/Multiplayer/Participants/ParticipantPanel.cs

```diff
@@ -162,6 +162,15 @@ protected override void OnRoomUpdated()
 
             const double fade_time = 50;
 
+            var ruleset = rulesets.GetRuleset(Room.Settings.RulesetID);
+
+            var currentModeRank = User.User?.GetStatisticsFor(ruleset)?.GlobalRank;
+
+            // fallback to current mode rank for testing purposes.
+            currentModeRank ??= User.User?.CurrentModeRank;
```

Comment:
why? can't the tests be fixed? leaving this is potentially going to make the incorrect rank for a user show if they have never played in the ruleset of the room they are participating in.

Reply:
hmm, yeah I was unsure about it, maybe making `User` expose the statistics-per-ruleset dictionary for tests to add custom ranks to.